### ISSUE TITLE
Fix S3900 FP: Coalesce assignment on static read-only members

### ISSUE THREAD

**Issue by Corniel** [SonarSource/sonar-dotnet#8541]
### Description

Rule S3900 considers the result of `CultureInfo.CurrentCulture` (and other) potentially `null`. 

### Repro steps
``` C#
static NumberFormatInfo OnField(CultureInfo? culture)
{
    culture ??= CultureInfo.CurrentCulture;
    return culture.NumberFormat; // FP, Current Culture will never be null.
}

static NumberFormatInfo OnMethod(CultureInfo? culture)
{
   culture ??= FallBack();
   return culture.NumberFormat; // FP, Current Culture will never be null.
}

static NumberFormatInfo WhenSuppressed(CultureInfo? culture)
{
   culture ??= UnkwownSource();
   return culture!.NumberFormat;  // FP, nullabillity is suppressed.
}

static CultureInfo FallBack() => CultureInfo.InvariantCulture;

static CultureInfo? UnkwownSource() => null;

```
When using ctor things work fine:

``` C#
static NumberFormatInfo OnObject(CultureInfo? culture)
{
    culture ??= new("en"); // Compliant
    return culture.NumberFormat;
}
```

It seems that nullable annotations are not considered, nor that the validation can be suppressed with `!`.

**Comment by pavel-mikula-sonarsource** [comment]
Hi @Corniel,

Thank you for reporting this, I confirm it as FP.

We should be able to tell that `CultureInfo.CurrentCulture` is never `null`.
We don't have cross-procedure capability yet in our engine, so the `FallBack()` case will be fixed soon. We'll hopefully be able to fix that in a year.

**Comment by Corniel** [comment]
@pavel-mikula-sonarsource I'm looking forward to the improvements.

**Comment by martin-strecker-sonarsource** [comment]
The issue is tracked by https://sonarsource.atlassian.net/browse/NET-1206 now.
